### PR TITLE
Add auth controller/factory and signing signup views

### PR DESCRIPTION
**Pull request by mwharris1313**
Add auth controller/factory and signing signup views

### CODE REVIEW

**Comment by jyeg** [comment]
Mike, will the there be a problem since you have an ng-model and an ngController in each your signin.html and signup.html?  if it works then nevermind.

File: client/app/auth/auth.js [hunk 1]
```diff
@@ -1,4 +1,48 @@
-angular.module('courageousTrapeze.auth', [])
+var app = angular.module('courageousTrapeze.auth', []);
```
Comment (kevhuang) [inline client/app/auth/auth.js:1]: @mwharris333 Let's avoid using the variable name, `app`, for this auth module. The reason being, `app` is usually for the main module (defined in app.js). If you want to use a variable name for this auth module, perhaps a better variable name could be `auth` or `authModule`.

Another approach is to not use a variable and instead chain the controller like so:

```
angular.module('courageousTrapeze.auth', [])
.controller('AuthController', ['$scope', 'Auth', function($scope, Auth) {
  ...
}]);
```

File: client/app/auth/auth.js [hunk 2]
```diff
@@ -1,4 +1,48 @@
...
+    })
+  };
 
-.controller('AuthController', ['$scope', 'Auth', function($scope, Auth) {
 }]);
+
+app.factory('Auth', function($http){
+  var signup = function(count, cb){
+    return $http({
+      method: 'Post',
```
Comment (kevhuang) [inline client/app/auth/auth.js:25]: Does it matter that `Post` is title case as opposed to all upper case?

File: client/app/auth/auth.js [hunk 3]
```diff
@@ -1,4 +1,48 @@
...
+      url: '/api/user/signup'
+    })
+    .then(function (resp) {
+      console.log('RESPONSE:', resp)
+    });
+  };
+
+  var signin = function(count, cb){
+    return $http({
+      method: 'Post',
```
Comment (kevhuang) [inline client/app/auth/auth.js:35]: Should `Post` be all upper case? Does it matter?

File: client/app/auth/auth.js [hunk 4]
```diff
@@ -1,4 +1,48 @@
-angular.module('courageousTrapeze.auth', [])
+var app = angular.module('courageousTrapeze.auth', []);
+app.controller('AuthController', ['$scope', 'Auth', function($scope, Auth){
+  // var test = Counter();
+  $scope.user = {};
+  $scope.user.username = '';
+  $scope.user.password = '';
+
+  $scope.signin = function(count){
```
Comment (kevhuang) [inline client/app/auth/auth.js:8]: I noticed there are a couple of instances that the variable `count` shows up, but I'm not sure what it's used for. Can you please clarify?

**Comment by kevhuang** [comment]
@mwharris333 Also, I already included an Auth factory in client/app/factories/factories.js. I was thinking we could keep all the factories in that file so that it would be more modularized in that other modules and controllers can use any of the factories.

If you think otherwise, I'm open to hearing your opinion. In either case, can you refactor the codebase so that we don't have duplicate Auth factories?

**Pull request closed without merging**